### PR TITLE
User management interface

### PR DESCRIPTION
**Pull request by asportnoy**
This adds an interface to manage users on the admin page. All CRUD operations are implemented.

This also fixes some stuff with the API, however most of the changes are on the frontend.

This uses the API routes from #51 and completes part of #37.

### CODE REVIEW

**Comment by asportnoy** [comment]
Note: editing can cause call stack exceeding, likely due to circular reference. Happens when you change your balance expires to a date in the past or change someone from a student to a staff.

**Comment by asportnoy** [comment]
thanks matt nice commit

**Comment by mswider** [comment]
> thanks matt nice commit

tysm bestie

**Comment by asportnoy** [comment]
Previous issue is fixed but there is a time zone issue
image

**Comment by mswider** [comment]
> Previous issue is fixed but there is a time zone issue

What is the issue?

**Comment by asportnoy** [comment]
Nevermind. The issue was that the time was resetting in UTC as opposed to local time but it was caused by the docker container using UTC on the codETHS server. I edited some config and it now works as expected.

File: frontend/src/pages/admin/index.svelte [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+			method: 'DELETE',
+		}).catch(() => null);
+
+		submitStatus = res && res.ok ? 'SUCCESS' : 'ERROR';
+		if (submitStatus == 'SUCCESS') {
+			toastContainer.toast(`${selectedStudent.text} deleted`, 'success');
+			studentSearch.el.setValue(null, null);
+			setData();
+		} else {
+			toastContainer.toast('Error deleting item.', 'error');
```
Comment (WZMorton) [inline frontend/src/pages/admin/index.svelte]: Shouldn't this say "Error deleting user." rather than "Error deleting item."

Comment (asportnoy) [reply frontend/src/pages/admin/index.svelte]: Copy paste moment

**Review by WZMorton — APPROVED** [review]

File: src/webserver/index.ts [hunk 2]
```diff
@@ -133,6 +133,7 @@ app.get(
 	(req, res) => {
 		if (req.user) {
 			if (req.user.hasRole('STAFF')) return res.redirect('/staff');
+			else if (req.user.hasRole('ADMIN')) return res.redirect('/admin');
 			else if (req.user.hasRole('STUDENT'))
```
Comment (mswider) [inline src/webserver/index.ts:137]: Should this be done in the order of highest to lowest permissions? (Admin, staff, then student)

**Review by mswider — APPROVED** [review]
Looks good, I haven't found any more issues

**Pull request merged**